torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data."""
+import glob
+import os
+import re
+from typing import Any, Callable, Dict, List, Optional, Sequence, cast
+
+import matplotlib.pyplot as plt
+import torch
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class L7Irish(RasterDataset):
+    """L7 Irish dataset.
+
+    The `L7 Irish <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset # noqa: E501
+    is based on Landsat 7 Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes.
+
+    Dataset features:
+
+    * 206 scenes from Landsat-7 ETM+ tiles
+    * Imagery from global tiles between June 2000--December 2001
+    * 9 Level-1 spectral bands with 15, 30 and 60 m per pixel resolution
+
+    Dataset format:
+
+    * Images are composed of multiple single channel geotiffs
+    * Labels are multiclass, stored in a single geotiffs file per image
+    * Level-1 metadata (MTL.txt file)
+    * Landsat-7 ETM+ bands: (B10, B20, B30, B40, B50, B61, B62, B70, B80)
+
+    Dataset classes (5):
+
+    0. Fill
+    64. Cloud Shadow
+    128. Clear
+    192. Thin Cloud
+    255. Cloud
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.5066/F7XD0ZWC
+    * https://doi.org/10.1109/TGRS.2011.2164087
+    """
+
+    url = "https://huggingface.co/datasets/torchgeo/l7irish/resolve/main/{}.tar.gz"  # noqa: E501
+
+    md5s = {
+        "austral": "dbb6b5628f50861b9b89f548d25a925f",
+        "boreal": "cecc72de09aacde4c4f8d7f0cf0d3f6f",
+        "mid_latitude_north": "0f8382ca6554fb7cf9aff42226a14f9d",
+        "mid_latitude_south": "b17cf6d023f752c533211fdb742f296b",
+        "polar_north": "73923dcaf1b9b79bad82de1aa0740d1e",
+        "polar_south": "3bc9f4c6f8955b10b4d55d23e0ab2da7",
+        "subtropical_north": "f8f039970256902e6e9ebd6747589294",
+        "subtropical_south": "8346d73a983396c5d41b577c3a94bc26",
+        "tropical": "abe19b22b5d031e6b609cc7207706c3d",
+    }
+
+    classes = ["Fill", "Cloud Shadow", "Clear", "Thin Cloud", "Cloud"]
+
+    filename_glob = "L7*.TIF"
+
+    filename_regex = r"""
+        ^L7[12]
+        (?P<wrs_path>\d{3})
+        (?P<wrs_row>\d{3})
+        _(?P=wrs_row)
+        (?P<date>\d{8})
+        _(?P<band>B\d{2})
+        \.TIF$
+    """
+    date_format = "%Y%m%d"
+
+    separate_files = True
+    rgb_bands = ["B40", "B20", "B30"]
```

Comment:
```suggestion
    rgb_bands = ["B30", "B20", "B10"]
```
See https://developers.google.com/earth-engine/datasets/catalog/LANDSAT_LE07_C02_T1_TOA